File: Backend/TopSys.TopConWeb.Domain/Entities/DemaisServicos.cs
using System;
using TopSys.TopConWeb.Domain.Enums;
using static TopSys.TopConWeb.SharedKernel.Helpers.StringHelper;

namespace TopSys.TopConWeb.Domain.Entities
{
    public class DemaisServicos
    {
        public int Codigo { get; set; }

        public int UsinaCodigo { get; set; }
        public virtual Usina Usina { get; set; }

        public string MercadoriaCodigo { get; set; }
        public virtual Mercadoria Mercadoria { get; set; }

        public string UnidadeSigla { get; set; }
        public virtual Unidade Unidade { get; set; }

        public int NumeroDeCasasDecimais { get; set; }

        public float PrecoSugerido { get; set; }

        public float PrecoMinimo { get; set; }

       
        public string FrequenciaDeCobrancaString
        {
            get { return FrequenciaDeCobranca.ToString(); }
            private set { FrequenciaDeCobranca = EnumExtensions.ParseEnum<EFrequenciaDeCobranca>(value); }
        } 
        public EFrequenciaDeCobranca FrequenciaDeCobranca { get; set; }


        public string FormaDeCobrancaString
        {
            get { return FormaDeCobranca.ToString(); }
            private set { FormaDeCobranca = EnumExtensions.ParseEnum<EFormaDeCobrancaDemaisServicos>(value); }
        }
        public EFormaDeCobrancaDemaisServicos FormaDeCobranca { get; set; }

        public bool AtualizaEstoque { get; set; }

    }
}

File: Frontend/TopConWeb-AngularV8/src/app/classes/demais-servicos/demais-servicos.ts
import { Usina } from '../usina/usina';
import { Mercadoria, Unidade } from '../mercadoria/mercadoria';

export enum EFrequenciaDeCobranca {
	Contrato = 1,
	Programacao,
	Remessa, 
	M3,
	M3Bombeado,
	Bombeamento
}

export enum EFormaDeCobrancaDemaisServicos {
	NaRemessa = 1,
	FinalConcretagem,
}

export class DemaisServicos {
    codigo: number = 0;
    usina: Usina;
    mercadoria: Mercadoria;
    unidade: Unidade;
    numeroDeCasasDecimais: number = 0;
    precoSugerido: number = 0;
    precoMinimo: number = 0;
    frequenciaDeCobranca: EFrequenciaDeCobranca;
    formaDeCobranca: EFormaDeCobrancaDemaisServicos = EFormaDeCobrancaDemaisServicos.NaRemessa;
    atualizaEstoque: string = '';
}